src/System.Text.Json/src/System/Text/Json/Serialization/WriteStack.cs

```diff
@@ -71,5 +70,51 @@ public void Pop()
             Debug.Assert(_index > 0);
             Current = _previous[--_index];
         }
+
+        // Return a property path as a simple JSONPath using dot-notation when possible. When special characters are present, bracket-notation is used:
+        // $.x.y.z
+        // $['PropertyName.With.Special.Chars']
+        public string PropertyPath
+        {
+            get
+            {
+                StringBuilder sb = new StringBuilder("$");
```

Comment:
How often do we expect this to get called in running code? We should at least give this a reasonable initial capacity. If we call it a lot (I don't think so?) we should consider caching the builder.

Reply:
This is only called in exception cases.